test/camera/camera_test.py

```diff
@@ -0,0 +1,448 @@
+
+
+    def test_pan_right_relative_beyond_bound(self):
+        camera = Camera(camera_type=RELATIVE)
+        camera.pan(-91)
+
+        self.assertEqual(camera._current_pan_angle, 0)
+
+
+    def test_tilt_up_absolute(self):
+        camera = Camera()
+        camera.tilt(90)
+
+        self.assertEqual(camera._current_tilt_angle, 90)
+
+
+    def test_tilt_up_absolute_at_bound(self):
+        camera = Camera()
+        camera.tilt(150)
+
+        self.assertEqual(camera._current_tilt_angle, 150)
+
+
+    def test_tilt_up_absolute_beyond_bound(self):
+        camera = Camera()
+        camera.tilt(151)
+
+        self.assertEqual(camera._current_tilt_angle, 150)
+
+
+    def test_tilt_down_absolute(self):
+        camera = Camera()
+        camera.tilt(90)
+        camera.tilt(60)
+
+        self.assertEqual(camera._current_tilt_angle, 60)
+
+
+    def test_tilt_down_absolute_at_bound(self):
+        camera = Camera()
+        camera.tilt(90)
+        camera.tilt(0)
+
+        self.assertEqual(camera._current_tilt_angle, 0)
+
+
+    def test_tilt_down_absolute_beyond_bound(self):
+        camera = Camera()
+        camera.tilt(90)
+        camera.tilt(-1)
+
+        self.assertEqual(camera._current_tilt_angle, 0)
+
+
+    def test_tilt_up_relative(self):
+        camera = Camera(camera_type=RELATIVE)
+        camera.tilt(90)
+
+        self.assertEqual(camera._current_tilt_angle, 90)
+
+
+    def test_tilt_up_relative_at_bound(self):
+        camera = Camera(camera_type=RELATIVE)
+        camera.tilt(150)
+
+        self.assertEqual(camera._current_tilt_angle, 150)
+
+
+    def test_tilt_up_relative_beyond_bound(self):
+        camera = Camera(camera_type=RELATIVE)
+        camera.tilt(151)
+
+        self.assertEqual(camera._current_tilt_angle, 150)
+
+
+    def test_tilt_down_relative(self):
+        camera = Camera(camera_type=RELATIVE)
+        camera.tilt(90)
+        camera.tilt(-30)
+
+        self.assertEqual(camera._current_tilt_angle, 60)
+
+
+    def test_tilt_down_relative_at_bound(self):
+        camera = Camera(camera_type=RELATIVE)
+        camera.tilt(90)
+        camera.tilt(-90)
+
+        self.assertEqual(camera._current_tilt_angle, 0)
+
+
+    def test_tilt_down_relative_beyond_bound(self):
+        camera = Camera(camera_type=RELATIVE)
+        camera.tilt(90)
+        camera.tilt(-91)
+
+        self.assertEqual(camera._current_tilt_angle, 0)
+
+
+    def test_get_pan_angle(self):
+        camera = Camera()
+        camera.pan(90)
+
+        self.assertEqual(camera._current_pan_angle, camera.get_pan_angle())
+
+
+    def test_get_tilt_angle(self):
+        camera = Camera()
+        camera.tilt(90)
+
+        self.assertEqual(camera._current_tilt_angle, camera.get_tilt_angle())
+
+
+    def test_calibrate_pan_absolute_increase(self):
+        camera = Camera()
+        camera.calibrate_pan(10)
+        offset = int(self._db.get('pan_offset', default_value=0))
+
+        self.assertEqual(camera._pan_offset, 10)
```

Comment:
I think you could make this a function.

```python
def calibrate_pan_test(camera, calibration_value, expected_value):
    camera.calibrate_pan(calibration_value)
    offset = int(self._db.get('pan_offset', default_value=0))
    self.assertEqual(camera._pan_offset, expected_value)
    self.assertEqual(offset, expected_value)
```

Now you've reduced every test to two lines instead of 6

Reply:
Ah, was not sure if you could do that in testing. Much cleaner.